Made the project not needing ScienceMode if Rehastim2 is used

pariterre opened the pull request.

pariterre commented:
@aceglia if you are happy with these change, I can go back to lab to validate they are good for both Rehastim2 and RehastimP24 and then we can merge

EveCharbie commented:
@aceglia @Kev1CO @pariterre could this PR be merged please ?
I have the following error "NameError: name 'sciencemode' is not defined"

pariterre commented:
Since there was a lot of problems with using the library I moved almost all functions and abstracted the modules using interface, I added loggers, etc. But I am not maintainer on this project so I could not merge at the time. It seems this PR is now too much different with new developments to be manageably merged. 

As of now, I will not redo this work as Yosra do not need this module to evolve anymore from what we have already. If you need what I have done, you will have to use my branch, if you need the newest development, you will have to redo the work I have done.

Kev1CO commented:
For the integration of pysciencemode in pip and conda-forge I implemented your ideas of seperating both uses. It is less elegant than your PR but yet effective. After diagnostic with [EveCharbie](https://github.com/EveCharbie), the code error is not related to this PR. I will take your advice in closing this PR. Thank you for your time and work on pysciencemode.
